<?xml version="1.0"?> <launch>
  <include file="$(find lms1xx)/launch/LMS1xx.launch" />

  <node pkg="narrow_path" name="narrow_path_node" type="narrow_path_node" output="screen">
    <param name="CONST_SPEED" type="int"  value="5"/>
    <param name="CONST_STEER" type="int"  value="0"/>
    <param name="FILTER_RAVA_RADIUS" type="double"  value="0.17"/>
    <param name="STEER_WEIGHT" type="int"  value="40"/>
    <param name="ONE_SIDE_WEIGHT" type="double"  value="-30.0"/>
    <param name="DETECT_DISTANCE" type="int"  value="4"/>
  </node>

  <node pkg="platform_control" name="write_serial" type="write_serial" />
  

  <node pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen" name="laser_filter">
    <rosparam command="load" file="$(find narrow_path)/box_filter.yaml" />
  </node>
  
  <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_config_path)"/>
-->
  <node name="obstacle_extractor" pkg="obstacle_detector" type="obstacle_extractor_node">
	<remap from="scan" to="scan_filtered"/>
    <param name="active"               value="true"/>
    <param name="use_scan"             value="true"/>
    <param name="use_pcl"              value="false"/>

    <param name="use_split_and_merge"    value="true"/>
    <param name="circles_from_visibles"  value="true"/>
    <param name="discard_converted_segments" value="true"/>
    <param name="transform_coordinates"  value="true"/>

    <param name="min_group_points"     value="5"/>

    <param name="max_group_distance"   value="0.05"/>
    <param name="distance_proportion"  value="0.00628"/>
    <param name="max_split_distance"   value="0.2"/>
    <param name="max_merge_separation" value="0.2"/>
    <param name="max_merge_spread"     value="0.2"/>
    <param name="max_circle_radius"    value="0.5"/>
    <param name="radius_enlargement"   value="0.1"/>

    <param name="frame_id"             value="laser"/>
  </node>
<!--
  <node pkg="urg_node" name="urg_node" type="urg_node" output="screen">
    <param name="ip_address" value="192.168.0.11"/>
  </node>
-->

</launch>
